.checkpatch.conf

```diff
@@ -0,0 +1,19 @@
+--no-tree
+--strict
+--ignore=SPLIT_STRING
+--ignore SSCANF_TO_KSTRTO
+--ignore=NEW_TYPEDEFS
+--ignore=DEPRECATED_VARIABLE
+--ignore=COMPARISON_TO_NULL
+--ignore=BIT_MACRO
+--ignore=PREFER_PRINTF
+--ignore=PREFER_SCANF
+--ignore=VOLATILE
+--ignore=AVOID_EXTERNS
+--ignore=CONST_STRUCT
+--ignore=PREFER_KERNEL_TYPES
+--ignore=CONSTANT_COMPARISON
+--ignore=BLOCK_COMMENT_STYLE
+--ignore=UNNECESSARY_PARENTHESES
+--codespell
+--codespellfile=/usr/share/codespell/dictionary.txt
```

Comment:
What if the file doesn't exist? Will `checkpatch.pl` error out? If this adds a dependency on the `codespell` package we can always add it to the prerequisites of course.

Reply:
@jforissier that is optional thing but warning will be generated:

```
./scripts/checkpatch.sh HEAD
Checking commit(s):
e55cd0cb (HEAD -> checkpatch) move checkpatch options to config file
No codespell typos will be found - file '/usr/share/codespell/dictionary1.txt': No such file or directory
WARNING: A patch subject line should describe the change not the tool that found it
#4: 
Subject: [PATCH] move checkpatch options to config file

total: 0 errors, 1 warnings, 0 checks, 28 lines checked
muvarov@maximlap:/opt/Linaro/GIT/optee_os.git$ 
```

Reply:
OK thanks. Let's keep `--codespell` and drop `--codespellfile` since the path is already the default in `checkpatch.pl`.

As for:
```
WARNING: A patch subject line should describe the change not the tool that found it #4: 
Subject: [PATCH] move checkpatch options to config file
```
We will live with this warning, I can't imagine a better wording.